src/Gaffer/Animation.cpp

```diff
@@ -193,6 +206,34 @@ const Animation::CurvePlug *Animation::Key::parent() const
 	return m_parent;
 }
 
+Animation::Key::HookAccess::hook_ptr Animation::Key::HookAccess::to_hook_ptr( Animation::Key::HookAccess::value_type& value )
+{
+	return &( value.m_hook );
+}
+
+Animation::Key::HookAccess::const_hook_ptr Animation::Key::HookAccess::to_hook_ptr( const Animation::Key::HookAccess::value_type& value )
+{
+	return &( value.m_hook );
+}
+
+Animation::Key::HookAccess::pointer Animation::Key::HookAccess::to_value_ptr( Animation::Key::HookAccess::hook_ptr hook )
+{
+	return boost::intrusive::get_parent_from_member< Animation::Key >( hook, & Animation::Key::m_hook );
+}
+
+Animation::Key::HookAccess::const_pointer Animation::Key::HookAccess::to_value_ptr( Animation::Key::HookAccess::const_hook_ptr hook )
+{
+	return boost::intrusive::get_parent_from_member< Animation::Key >( hook, & Animation::Key::m_hook );
+}
+
+void Animation::Key::Dispose::operator()( Animation::Key* const key ) const
+{
+	assert( key != 0 );
+
+	key->m_parent = 0;
```

Comment:
Any reason this isn't ` = nullptr`?

Reply:
= 0 is valid C++, whats the benefit of nullptr ?

Reply:
At a glance, ` = nullptr` is a pointer assignment, and ` = 0` is an integer assignment. There's plenty of valid C++ that isn't necessarily the clearest way of expressing something.